src/OpenTelemetry.Exporter.Prometheus/Implementation/PrometheusSerializer.cs

```diff
@@ -249,21 +249,42 @@ public static int WriteMetricName(byte[] buffer, int cursor, string metricName)
                         buffer[cursor++] = unchecked((byte)'_');
                         break;
                     default:
-                        cursor = WriteUnicodeNoEscape(buffer, cursor, ordinal);
+                        buffer[cursor++] = unchecked((byte)ordinal);
                         break;
                 }
             }
 
+            if (!string.IsNullOrEmpty(metricUnit))
+            {
+                buffer[cursor++] = unchecked((byte)'_');
+
+                for (int i = 0; i < metricUnit.Length; i++)
+                {
+                    var ordinal = (ushort)metricUnit[i];
+
+                    if ((ordinal >= (ushort)'A' && ordinal <= (ushort)'Z') ||
+                        (ordinal >= (ushort)'a' && ordinal <= (ushort)'z') ||
+                        (ordinal >= (ushort)'0' && ordinal <= (ushort)'9'))
+                    {
+                        buffer[cursor++] = unchecked((byte)ordinal);
+                    }
+                    else
+                    {
+                        buffer[cursor++] = unchecked((byte)'_');
+                    }
+                }
+            }
+
             return cursor;
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static int WriteHelpText(byte[] buffer, int cursor, string metricName, string metricDescription = null)
+        public static int WriteHelpText(byte[] buffer, int cursor, string metricName, string metricUnit = null, string metricDescription = null)
         {
             cursor = WriteAsciiStringNoEscape(buffer, cursor, "# HELP ");
-            cursor = WriteMetricName(buffer, cursor, metricName);
+            cursor = WriteMetricName(buffer, cursor, metricName, metricUnit);
 
-            if (metricDescription != null)
+            if (!string.IsNullOrEmpty(metricDescription))
```

Comment:
This is to align with the OpenTelemetry Metrics spec:

_If the description is not provided or the description is null, the API and SDK MUST make sure that the behavior is the same as an empty description string._

Reply:
@cijothomas I wonder if the SDK guarantees that or we need to cover it here?

Reply:
No. We should handle it in SDK itself, so exporters need not repeat this.

Reply:
Got it, I take it as:
1. The SDK needs to convert null description to empty string.
2. We can keep the logic here since the perf difference between `string.IsNullOrEmpty` and the empty string check is unnoticeable.